lib/matplotlib/figure.py

```diff
@@ -48,69 +49,42 @@ def _stale_figure_callback(self, val):
         self.figure.stale = val
 
 
-class _AxesStack(cbook.Stack):
+class _AxesStack:
     """
-    Specialization of Stack, to handle all tracking of Axes in a Figure.
+    Helper class to track axes in a figure.
 
-    This stack stores ``ind, axes`` pairs, where ``ind`` is a serial index
-    tracking the order in which axes were added.
-
-    AxesStack is a callable; calling it returns the current axes.
+    Axes are tracked both in the order in which they have been added (`._axes`
+    iteration order) and in the separate "gca" stack (which is the index to
```

Comment:
```suggestion
    insertion/iteration order) and in the separate "gca" stack (which is the index to
```
It's helpful to mention "insertion order" in this context.